rpc-interface/src/types.rs

```diff
@@ -281,12 +281,32 @@ impl Block {
                                     .clone()
                                     .into_iter()
                                     .map(|tx| {
-                                        Transaction::from_blockchain(
-                                            tx,
-                                            block_number,
-                                            timestamp,
-                                            head_height,
-                                        )
+                                        // We obtain an internal executed transaction
+                                        // We need to grab the internal transaction and map it to the RPC transaction structure
+                                        match tx {
```

Comment:
I think you can replace this with the `from_blockchain` method of the `Executed Transaction` type? There the two states are already differentiated.